atlp-rwanda/hackers-ec-be
# #187419081 Not appear expired products on available list.

## Pull Request Description
### Title: Implementing Product Expiration Lifecycle
### What does this PR do?
This PR ensures that expired products do not appear in the list of available products. It introduces a mechanism to automatically mark products as 'Unavailable' once their expiration date has passed.

### Description of Task to be completed?
The task involves the following:

Updating the product status to 'Unavailable' once the expiration date has passed.
Ensuring that the product does not appear in the list of available products once it's marked as 'Unavailable'.
How should this be manually tested?
To manually test this feature, follow these steps:

Insert CLONE_TIME=0 0 * * * in your .env file. This sets the cron job to run every mid night for testing purposes.
Run the program using npm run dev.
Observe that products with passed expiration dates are marked as 'Unavailable' and are removed from the list of available products.
